### PR TITLE
B1603: Abstracted creation of temp directories for a couple of test classes

### PR DESCRIPTION
**Pull request by andrewbp**
I did see some other classes that use System.getProperty, but they necessarily need "/tmp" at the moment. Is the scope of this ticket to change how those classes work as well?

### CODE REVIEW

**Comment by andrewbp** [comment]
https://github.com/NationalSecurityAgency/burrito-grande/issues/1603

**Comment by jpdahlke** [comment]
Does this have the ability to utilize the '@TempDir' annotation now that we are moving to junit 5?

https://github.com/NationalSecurityAgency/emissary/blob/master/src/test/java/emissary/command/FeedCommandIT.java#L49

File: src/main/java/emissary/test/core/UnitTest.java [hunk 1]
```diff
@@ -91,7 +107,7 @@ public void evaluate() throws Throwable {
     protected String origConfigPkg = null;
 
     // TODO: remove this and see what breaks
```
Comment (cfkoehler) [inline src/main/java/emissary/test/core/UnitTest.java:109]: Do you also want to remove the comment?

Comment (cfkoehler) [inline src/main/java/emissary/test/core/UnitTest.java:109]: Do you also want to remove the comment?

File: src/test/java/emissary/util/shell/ExecutrixTest.java [hunk 2]
```diff
@@ -23,35 +23,33 @@
...
 import emissary.test.core.UnitTest;
+import org.apache.commons.lang.SystemUtils;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 
 public class ExecutrixTest extends UnitTest {
     private Executrix e;
-    private boolean isWindows = System.getProperty("os.name").indexOf("Window") != -1;
+    private final boolean isWindows = SystemUtils.OS_NAME.contains("Windows");
```
Comment (cfkoehler) [inline src/test/java/emissary/util/shell/ExecutrixTest.java:33]: Conflict with the removal of windows PR

File: src/test/java/emissary/util/shell/ExecutrixTest.java [hunk 3]
```diff
@@ -23,35 +23,33 @@
...
     public void setUp() throws Exception {
         this.e = new Executrix();
 
-        final String tmp = System.getProperty("java.io.tmpdir");
-        if (tmp == null || (tmp.indexOf("~") > -1 && this.isWindows)) {
-            if (this.isWindows) {
-                File f = new File("c:/tmp");
+        final String tmp = TMPDIR;
+        System.out.println(tmp);
+        if ((tmp.contains("~") && this.isWindows)) {
```
Comment (cfkoehler) [inline src/test/java/emissary/util/shell/ExecutrixTest.java:42]: Also conflicting with windows removal

**Review by cfkoehler — CHANGES_REQUESTED** [review]

File: src/test/java/emissary/util/shell/ExecutrixTest.java (hunk 2 above)
Comment (andrewbp) [reply src/test/java/emissary/util/shell/ExecutrixTest.java:33]: Ah, I wasn't aware of that effort. I'll make changes.

**Comment by andrewbp** [comment]
References to Windows removed.

**Comment by cfkoehler** [comment]
@andrewbp Can you rebase and resolve conflicts in InitTest.java when you get a chance.

File: src/test/java/emissary/util/shell/ExecutrixTest.java (hunk 2 above)
Comment (jpdahlke) [reply src/test/java/emissary/util/shell/ExecutrixTest.java]: That effort is an emissary 8 effort due to breaking compatibility. This PR was slated for the 7 line. If anything, the windows support removal PR would need to rebase if something done in this PR conflicted. This effort should not remove windows support.

**Review by jpdahlke — CHANGES_REQUESTED** [review]
Please rebase and restore windows support.

File: src/test/java/emissary/util/shell/ExecutrixTest.java (hunk 2 above)
Comment (cfkoehler) [reply src/test/java/emissary/util/shell/ExecutrixTest.java]: Whoops. My fault @andrewbp. Sorry about that

**Comment by andrewbp** [comment]
Reverted Windows support removal and rebased.

File: src/test/java/emissary/test/core/UnitTest.java [hunk 4]
```diff
@@ -82,6 +84,20 @@ public void evaluate() throws Throwable {
...
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    protected String TMPDIR = "/tmp";
+
+    @Before
+    public void setupTmpDir() {
+        try {
+            TMPDIR = temporaryFolder.newFolder().getAbsolutePath();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (dev-mlb) [inline src/test/java/emissary/test/core/UnitTest.java:96]: use logger

File: src/test/java/emissary/util/shell/ExecutrixTest.java [hunk 5]
```diff
@@ -24,35 +24,33 @@
...
     @BeforeEach
     public void setUp() throws Exception {
         this.e = new Executrix();
 
-        final String tmp = System.getProperty("java.io.tmpdir");
-        if (tmp == null || (tmp.contains("~") && this.isWindows)) {
-            if (this.isWindows) {
-                File f = new File("c:/tmp");
+        final String tmp = TMPDIR;
+        System.out.println(tmp);
```
Comment (dev-mlb) [inline src/test/java/emissary/util/shell/ExecutrixTest.java:42]: remove println

File: src/test/java/emissary/util/shell/ExecutrixTest.java [hunk 6]
```diff
@@ -24,35 +24,33 @@
...
+        if ((tmp.contains("~") && this.isWindows)) {
```
Comment (dev-mlb) [inline src/test/java/emissary/util/shell/ExecutrixTest.java:43]: You probably can just remove this whole code block now. Since junit is setting the temporary dir, it is unlikely that we'll get a ~ in a windows path.

File: src/test/java/emissary/util/shell/ExecutrixTest.java [hunk 7]
```diff
@@ -290,16 +283,16 @@ void testExecute() throws IOException {
 
         final String cyg = System.getProperty("CYGHOME");
         final boolean cyghome = cyg != null && cyg.contains(":");
-        final String cmd = (this.isWindows ? (cyghome ? "/bin/cp" : "copy") : "cp") + " <INPUT_NAME> <OUTPUT_NAME>";
+        final String cmd = "cp <INPUT_NAME> <OUTPUT_NAME>";
```
Comment (dev-mlb) [inline src/test/java/emissary/util/shell/ExecutrixTest.java:286]: revert this change

File: src/test/java/emissary/util/shell/ExecutrixTest.java [hunk 8]
```diff
@@ -290,16 +283,16 @@ void testExecute() throws IOException {
...
         String[] c = this.e.getCommand(cmd, names);
         assertNotNull(c, "Command returned");
-        assertEquals((this.isWindows ? "cmd" : "/bin/sh"), c[0], "Command runner");
+        assertEquals("/bin/sh", c[0], "Command runner");
```
Comment (dev-mlb) [inline src/test/java/emissary/util/shell/ExecutrixTest.java:289]: revert this change

File: src/test/java/emissary/util/shell/ExecutrixTest.java [hunk 9]
```diff
@@ -290,16 +283,16 @@ void testExecute() throws IOException {
...
 
         this.e.setCommand(cmd);
         c = this.e.getCommand(names);
         assertNotNull(c, "Command returned");
-        assertEquals((this.isWindows ? "cmd" : "/bin/sh"), c[0], "Command runner");
 
+        assertEquals("/bin/sh", c[0], "Command runner");
```
Comment (dev-mlb) [inline src/test/java/emissary/util/shell/ExecutrixTest.java:295]: revert

**Review by dev-mlb — CHANGES_REQUESTED** [review]

**Comment by andrewbp** [comment]
I made the requested changes but I think I need to redo the rebase.

**Comment by andrewbp** [comment]
OK! All good now!

File: src/test/java/emissary/test/core/UnitTest.java [hunk 10]
```diff
@@ -128,20 +140,23 @@ public synchronized void setupSystemProperties() {
      * Configure the test stuff
      * <p>
      * Beware though, if you use @BeforeClass this will not have had a change to run. So you can do something like
+<<<<<<< HEAD
```
Comment (cfkoehler) [inline src/test/java/emissary/test/core/UnitTest.java:143]: Need to complete the merge to remove the conflict markers.

File: src/test/java/emissary/util/shell/ExecutrixTest.java [hunk 11]
```diff
@@ -24,37 +24,19 @@
...
 
 import emissary.test.core.UnitTest;
+import org.apache.commons.lang3.SystemUtils;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 class ExecutrixTest extends UnitTest {
     private Executrix e;
-    private final boolean isWindows = System.getProperty("os.name").contains("Window");
```
Comment (cfkoehler) [inline src/test/java/emissary/util/shell/ExecutrixTest.java:33]: I know this is on me. But should this be reverted to match the original. Then Emissary 8 can remove all the windows references? Not aware of the difference between System.getProperty and SystemUtils.os_name

**Review by cfkoehler — CHANGES_REQUESTED** [review]

File: src/test/java/emissary/util/shell/ExecutrixTest.java (hunk 11 above)
Comment (andrewbp) [reply src/test/java/emissary/util/shell/ExecutrixTest.java:33]: This change is part of what was requested in the ticket. From the ticket:


"We have quite a few test classes that rely on pulling the java.io.tmpdir value from System.getProperty.

UnitTest, ExecutrixTest, ...

I think it would be a good idea to move to using the Files.createTempFile or Files.createTempDirectory where possible to abstract us away from how temporary files are created (even if that behind the scenes leverages the same call)."

Comment (andrewbp) [reply src/test/java/emissary/util/shell/ExecutrixTest.java:33]: Actually, reading it again, I may have incorrectly interpreted the comment as "we should stop using System.getProperty". I can revert the change if that's preferred.

File: src/test/java/emissary/util/shell/ExecutrixTest.java [hunk 12]
```diff
@@ -24,37 +24,19 @@
...
+    private final boolean isWindows = SystemUtils.OS_NAME.contains("Windows");
 
     @Override
     @BeforeEach
     public void setUp() throws Exception {
         this.e = new Executrix();
```
Comment (dev-mlb) [inline src/test/java/emissary/util/shell/ExecutrixTest.java:39]: I think you may still have to set `this.e.setTmpDir `but you no longer need to check for `~` in a window's path

File: src/test/java/emissary/test/core/UnitTest.java [hunk 13]
```diff
@@ -128,20 +140,17 @@ public synchronized void setupSystemProperties() {
...
     protected void configure() {
         thisPackage = this.getClass().getPackage();
         setupSystemProperties();
     }
 
-    @Before
-    public void setUp() throws Exception {
-
-    }
+    public void setUp() throws Exception {}
```
Comment (dev-mlb) [inline src/test/java/emissary/test/core/UnitTest.java:153]: Why this change?

**Review by dev-mlb — CHANGES_REQUESTED** [review]

File: src/test/java/emissary/test/core/UnitTest.java (hunk 13 above)
Comment (jpdahlke) [reply src/test/java/emissary/test/core/UnitTest.java:153]: As in the removal of the @Before?

**Review by jpdahlke — APPROVED** [review]
